### PR TITLE
Add RuntimeContext test part 2 (of many)

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Api/Context/RuntimeContext.cs [hunk 1]
```diff
@@ -67,7 +67,14 @@ public static RuntimeContextSlot<T> RegisterSlot<T>(string name)
...
         public static RuntimeContextSlot<T> GetSlot<T>(string name)
         {
-            return (RuntimeContextSlot<T>)Slots[name];
+            if (Slots.TryGetValue(name, out var slot) && slot is RuntimeContextSlot<T> expectedSlot)
+            {
+                return expectedSlot;
+            }
+            else
+            {
+                return null;
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Context/RuntimeContext.cs:76]: I think this should be a throw. Or the method should be TryGetSlot?

File: src/OpenTelemetry.Api/Context/RuntimeContext.cs [hunk 2]
```diff
@@ -67,7 +67,14 @@ public static RuntimeContextSlot<T> RegisterSlot<T>(string name)
         /// <returns>The slot previously registered.</returns>
         public static RuntimeContextSlot<T> GetSlot<T>(string name)
         {
-            return (RuntimeContextSlot<T>)Slots[name];
+            if (Slots.TryGetValue(name, out var slot) && slot is RuntimeContextSlot<T> expectedSlot)
+            {
+                return expectedSlot;
+            }
+            else
```
Comment (pjanotti) [inline src/OpenTelemetry.Api/Context/RuntimeContext.cs:74]: nit: either with `TryGetSlot` or `GetSlot` you can get rid of the `else` statement

File: src/OpenTelemetry.Api/Context/RuntimeContext.cs [hunk 3]
```diff
@@ -67,7 +67,12 @@ public static RuntimeContextSlot<T> RegisterSlot<T>(string name)
         /// <returns>The slot previously registered.</returns>
         public static RuntimeContextSlot<T> GetSlot<T>(string name)
         {
-            return (RuntimeContextSlot<T>)Slots[name];
+            if (Slots.TryGetValue(name, out var slot) && slot is RuntimeContextSlot<T> expectedSlot)
+            {
+                return expectedSlot;
+            }
+
+            throw new Exception("Slot with given name not found");
```
Comment (reyang) [inline src/OpenTelemetry.Api/Context/RuntimeContext.cs:75]:
```suggestion
```suggestion
            Slots.TryGetValue(name, out var slot);
            return slot as RuntimeContextSlot<T> ?? throw new Exception("Slot with given name not found");
```

Comment (reyang) [reply src/OpenTelemetry.Api/Context/RuntimeContext.cs:75]: And we probably want to use `KeyNotFoundException` instead of a general one?

Comment (cijothomas) [reply src/OpenTelemetry.Api/Context/RuntimeContext.cs:75]: yes. agree

Comment (reyang) [reply src/OpenTelemetry.Api/Context/RuntimeContext.cs:75]: E.g. `throw new KeyNotFoundException(name)`.

BTW, we probably also need to check if key is null or empty string.

File: src/OpenTelemetry.Api/Context/RuntimeContext.cs [hunk 4]
```diff
@@ -39,35 +39,46 @@ public static class RuntimeContext
         /// <summary>
         /// Register a named context slot.
         /// </summary>
-        /// <param name="name">The name of the context slot.</param>
+        /// <param name="slotName">The name of the context slot.</param>
```
Comment (reyang) [inline src/OpenTelemetry.Api/Context/RuntimeContext.cs:42]: I don't really know if this is idiomatic to C# or general OOP.
Let's say we have something like this:

```csharp
class Horse
{
    private string name;
    private int age;
    public Person(string name, int age)
    {
    }
}
```

or this:

```csharp
class Horse
{
    private string horseName;
    private int horseAge;
    public Horse(string horseName, int horseAge)
    {
    }
}
```

**Pull request merged**